cmd/dns.go

```diff
@@ -0,0 +1,71 @@
+package cmd
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+func dnsCommand() *cobra.Command {
+	versionCmd := &cobra.Command{
+		Use:   "dns",
+		Short: "DNS Operator command line utility",
+		Long:  "DNS Operator command line utility",
+		RunE:  runDNS,
+	}
+	return versionCmd
+}
+
+func runDNS(cmd *cobra.Command, args []string) error {
+	// pass verbose from root
+	args = append(args, fmt.Sprintf("--verbose=%t", verbose))
+
+	err := execPlugin(args)
+	if err != nil {
+		if strings.Contains(err.Error(), "executable file not found in $PATH") {
+			err = getDNSplugin(cmd)
+			if err != nil {
+				return fmt.Errorf("failed to get dns plugin: %w", err)
+			}
+			err = execPlugin(args)
+		}
+	}
+	return err
+}
+
+func getDNSplugin(cmd *cobra.Command) error {
+	fmt.Println("DNS plugin is not found in PATH\n" +
+		"Would you like to get DNS plugin? [Y/N]\n" +
+		"defaults to N")
+
+	reader := bufio.NewReader(cmd.InOrStdin())
+	text, _ := reader.ReadString('\n')
+	text = strings.TrimSuffix(text, "\n")
+
+	if strings.ToLower(text) == "y" {
+		return getPlugin()
+	}
+	fmt.Println("Consider getting plugin into PATH\n" +
+		"You can get it from the dns-operator repository:\n" +
+		"https://github.com/Kuadrant/dns-operator")
+	return nil
+}
+
+func execPlugin(args []string) error {
+	out, err := exec.Command("kubectl-dns", args...).Output()
+	if err != nil {
+		return fmt.Errorf("failed to run dns plugin: %w", err)
+	}
+	fmt.Printf("%s\n", out)
+	return nil
+}
+
+func getPlugin() error {
```

Comment:
I think it makes more sense to just output this message rather than the question on line 41, I'd find that a little annoying a user.